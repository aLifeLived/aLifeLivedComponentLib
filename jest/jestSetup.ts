/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
