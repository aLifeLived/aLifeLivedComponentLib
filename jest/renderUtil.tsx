import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
