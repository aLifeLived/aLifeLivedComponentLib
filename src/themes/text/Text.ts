import { createText, ResponsiveValue } from '@shopify/restyle';
import { ColorThemeTokens } from '../palette';
import { Theme } from '../theme';
import { ValueOf } from '../../ts/utils';

const FontFamilies = {
  sfUiTextBold: 'SFUIText-Bold',
  sfUiTextRegular: 'SFUIText-Regular',
  sfUiTextSemiBold: 'SFUIText-Semibold',
  sfUiDisplayRegular: 'SFUIDisplay-Regular',
};

type TextVariantToken =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'intro'
  | 'body'
  | 'bodySmall'
  | 'bodyXSmall';

type TextVariant = {
  fontFamily: ValueOf<typeof FontFamilies>;
  fontSize: number | ResponsiveValue<number, Theme>;
  lineHeight?: number | string;
  color: ColorThemeTokens;
};

export const textVariants: Record<TextVariantToken, TextVariant> = {
  h1: {
    fontSize: 52,
    lineHeight: 52,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextBold,
  },
  h2: {
    fontSize: 32,
    lineHeight: 40,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextBold,
  },
  h3: {
    fontSize: 24,
    lineHeight: 28.8,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextBold,
  },
  h4: {
    fontSize: 20,
    lineHeight: 24,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextRegular,
  },
  intro: {
    fontSize: 16,
    lineHeight: 24,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextSemiBold,
  },
  body: {
    fontSize: 18,
    lineHeight: 20,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiDisplayRegular,
  },
  bodySmall: {
    fontSize: 16,
    lineHeight: 20,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiDisplayRegular,
  },
  bodyXSmall: {
    fontSize: 14,
    lineHeight: 20,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiDisplayRegular,
  },
};

export const Text = createText<Theme>();
