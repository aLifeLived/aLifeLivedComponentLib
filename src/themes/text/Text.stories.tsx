import React from 'react';
import { storiesOf } from '@storybook/react-native';

// Component
import { Text } from './Text';
import { CenterView } from '../../components/centerView/CenterView';

storiesOf('Text', module)
  .addDecorator(Story => <CenterView children={Story()} />)
  .add('Text', () => (
    <>
      <Text variant="h1">H1 Heading</Text>
      <Text variant="h2">H2 Heading</Text>
      <Text variant="h3">H3 Heading</Text>
      <Text variant="h4">H4 Heading</Text>
      <Text variant="body">Body</Text>
    </>
  ));
