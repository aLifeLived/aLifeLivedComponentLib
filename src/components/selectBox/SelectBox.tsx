import React, { ReactChild } from 'react';

// Components
import { Box } from '../../themes/box/Box';
import { SystemButton } from '../../themes/systemButton/SystemButton';
import { Text } from '../../themes/text/Text';

type SelectBoxType = {
  variant: 'horizontal' | 'vertical';
  title: string;
  subText: string;
  isSelected: boolean;
  onPress: () => void;
  children?: ReactChild;
};

export const SelectBox: React.FC<SelectBoxType> = ({
  variant,
  title,
  subText,
  isSelected,
  onPress,
  children,
}) => {
  return (
    <SystemButton onPress={() => onPress()}>
      <Box
        flexDirection={variant === 'horizontal' ? 'row' : 'column'}
        alignContent="center"
        alignItems="center"
        borderWidth={1}
        borderColor="borderColor"
        borderRadius="medium"
        backgroundColor={isSelected ? 'primary' : 'white'}
        boxShadow="md"
      >
        <Box margin="s">{children}</Box>
        <Box margin="s" alignContent="center" width="80%" flexWrap="nowrap">
          <Text
            variant="h3"
            fontWeight="bold"
            color="textContrastHigh"
            textAlign={variant === 'horizontal' ? 'left' : 'center'}
          >
            {title}
          </Text>
          <Text
            textAlign={variant === 'horizontal' ? 'left' : 'center'}
            marginTop="s"
            color="textContrastHigh"
            variant="caption"
          >
            {subText}
          </Text>
        </Box>
      </Box>
    </SystemButton>
  );
};
