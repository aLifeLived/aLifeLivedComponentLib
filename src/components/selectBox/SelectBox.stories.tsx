import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';
import { text, boolean } from '@storybook/addon-knobs';

// Component
import { SelectBox } from './SelectBox';
import { Box } from '../../themes/box/Box';
import { Icon } from '../icon/Icon';

storiesOf('Select box', module)
  .addDecorator(Story => Story())
  .add('Horizontal', () => (
    <Box margin="s">
      <SelectBox
        variant="horizontal"
        title={text('Title', 'Public')}
        subText={text(
          'Sub text',
          'By making your story public other users can listen and share it.'
        )}
        isSelected={boolean('isSelected', false)}
        onPress={action('Public pressed')}
      >
        <Icon icon="eye" color="contrastHigh" size={32} />
      </SelectBox>
    </Box>
  ))
  .add('Vertical', () => (
    <Box margin="s">
      <SelectBox
        variant="vertical"
        title={text('Title', 'Public')}
        subText={text(
          'Sub text',
          'By making your story public other users can listen and share it.'
        )}
        isSelected={boolean('isSelected', false)}
        onPress={action('Public pressed')}
      >
        <Icon icon="eye" color="contrastHigh" size={32} />
      </SelectBox>
    </Box>
  ));
