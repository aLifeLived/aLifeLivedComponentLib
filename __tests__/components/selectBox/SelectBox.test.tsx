import React, { ReactChild } from 'react';
import { render, fireEvent } from '../../../jest/renderUtil';

// Component
import { Text } from '../../../src/themes/text/Text';
import { SelectBox } from '../../../src/components/selectBox/SelectBox';

const mockOnPress = jest.fn();

type setupProps = {
  selectedValue: boolean;
  children?: ReactChild;
};

const setup = ({ selectedValue, children }: setupProps) => {
  const utils = render(
    <SelectBox
      title="Test title"
      onPress={mockOnPress}
      isSelected={selectedValue}
      subText="Test sub text"
    >
      {children}
    </SelectBox>
  );

  return { ...utils };
};

describe('components >> Select Box', () => {
  beforeEach(() => mockOnPress.mockClear());
  describe('Given select box is NOT selected and has no children', () => {
    const setupValues: setupProps = {
      selectedValue: false,
    };
    test('Should render snapshot correctly', () => {
      const { toJSON } = setup(setupValues);
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should find title and sub text', () => {
      const { getByText } = setup(setupValues);
      expect(getByText('Test title')).toBeDefined();
      expect(getByText('Test sub text')).toBeDefined();
    });

    test('Should fire on press', () => {
      const { getByText } = setup(setupValues);
      fireEvent.press(getByText('Test title'));
      expect(mockOnPress).toBeCalledTimes(1);
    });
  });

  describe('Given select box IS selected and has no children', () => {
    const setupValues: setupProps = {
      selectedValue: true,
    };
    test('Should render snapshot correctly', () => {
      const { toJSON } = setup(setupValues);
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should find title and sub text', () => {
      const { getByText } = setup(setupValues);
      expect(getByText('Test title')).toBeDefined();
      expect(getByText('Test sub text')).toBeDefined();
    });

    test('Should fire on press', () => {
      const { getByText } = setup(setupValues);
      fireEvent.press(getByText('Test title'));
      expect(mockOnPress).toBeCalledTimes(1);
    });
  });

  describe('Given select box IS selected and has children', () => {
    const setupValues: setupProps = {
      selectedValue: true,
      children: <Text>TEST CHILDREN</Text>,
    };
    test('Should render snapshot correctly', () => {
      const { toJSON } = setup(setupValues);
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should find title and sub text', () => {
      const { getByText } = setup(setupValues);
      expect(getByText('Test title')).toBeDefined();
      expect(getByText('Test sub text')).toBeDefined();
    });

    test('Should fire on press', () => {
      const { getByText } = setup(setupValues);
      fireEvent.press(getByText('Test title'));
      expect(mockOnPress).toBeCalledTimes(1);
    });
  });

  describe('Given select box is NOT selected and has children', () => {
    const setupValues: setupProps = {
      selectedValue: false,
      children: <Text>TEST CHILDREN</Text>,
    };
    test('Should render snapshot correctly', () => {
      const { toJSON } = setup(setupValues);
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should find title and sub text', () => {
      const { getByText } = setup(setupValues);
      expect(getByText('Test title')).toBeDefined();
      expect(getByText('Test sub text')).toBeDefined();
    });

    test('Should fire on press', () => {
      const { getByText } = setup(setupValues);
      fireEvent.press(getByText('Test title'));
      expect(mockOnPress).toBeCalledTimes(1);
    });
  });
});
