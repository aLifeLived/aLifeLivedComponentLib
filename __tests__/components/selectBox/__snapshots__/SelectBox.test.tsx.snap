// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components >> Select Box Given select box has a variant of horizontal Given select box IS selected and has children Should render snapshot correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "#0FCFA6",
          "borderColor": "#000000",
          "borderRadius": 8,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
        },
        Array [
          Object {
            "elevation": 2,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 4,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 10,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "margin": 8,
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "display": "flex",
            },
          ]
        }
      >
        TEST CHILDREN
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "display": "flex",
            "flexWrap": "nowrap",
            "margin": 8,
            "width": "80%",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 24,
              "textAlign": "left",
            },
          ]
        }
      >
        Test title
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 14,
              "lineHeight": 18,
              "marginTop": 8,
              "textAlign": "left",
            },
          ]
        }
      >
        Test sub text
      </Text>
    </View>
  </View>
</View>
`;

exports[`components >> Select Box Given select box has a variant of horizontal Given select box IS selected and has no children Should render snapshot correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "#0FCFA6",
          "borderColor": "#000000",
          "borderRadius": 8,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
        },
        Array [
          Object {
            "elevation": 2,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 4,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 10,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "margin": 8,
          },
          Array [
            Object {},
          ],
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "display": "flex",
            "flexWrap": "nowrap",
            "margin": 8,
            "width": "80%",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 24,
              "textAlign": "left",
            },
          ]
        }
      >
        Test title
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 14,
              "lineHeight": 18,
              "marginTop": 8,
              "textAlign": "left",
            },
          ]
        }
      >
        Test sub text
      </Text>
    </View>
  </View>
</View>
`;

exports[`components >> Select Box Given select box has a variant of horizontal Given select box is NOT selected and has children Should render snapshot correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "#f4f5f5",
          "borderColor": "#000000",
          "borderRadius": 8,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
        },
        Array [
          Object {
            "elevation": 2,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 4,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 10,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "margin": 8,
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "display": "flex",
            },
          ]
        }
      >
        TEST CHILDREN
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "display": "flex",
            "flexWrap": "nowrap",
            "margin": 8,
            "width": "80%",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 24,
              "textAlign": "left",
            },
          ]
        }
      >
        Test title
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 14,
              "lineHeight": 18,
              "marginTop": 8,
              "textAlign": "left",
            },
          ]
        }
      >
        Test sub text
      </Text>
    </View>
  </View>
</View>
`;

exports[`components >> Select Box Given select box has a variant of horizontal Given select box is NOT selected and has no children Should render snapshot correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "#f4f5f5",
          "borderColor": "#000000",
          "borderRadius": 8,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
        },
        Array [
          Object {
            "elevation": 2,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 4,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 10,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "margin": 8,
          },
          Array [
            Object {},
          ],
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "display": "flex",
            "flexWrap": "nowrap",
            "margin": 8,
            "width": "80%",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 24,
              "textAlign": "left",
            },
          ]
        }
      >
        Test title
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 14,
              "lineHeight": 18,
              "marginTop": 8,
              "textAlign": "left",
            },
          ]
        }
      >
        Test sub text
      </Text>
    </View>
  </View>
</View>
`;

exports[`components >> Select Box Given select box has a variant of vertical Given select box IS selected and has children Should render snapshot correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "#0FCFA6",
          "borderColor": "#000000",
          "borderRadius": 8,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "column",
        },
        Array [
          Object {
            "elevation": 2,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 4,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 10,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "margin": 8,
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "display": "flex",
            },
          ]
        }
      >
        TEST CHILDREN
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "display": "flex",
            "flexWrap": "nowrap",
            "margin": 8,
            "width": "80%",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 24,
              "textAlign": "center",
            },
          ]
        }
      >
        Test title
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 14,
              "lineHeight": 18,
              "marginTop": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Test sub text
      </Text>
    </View>
  </View>
</View>
`;

exports[`components >> Select Box Given select box has a variant of vertical Given select box IS selected and has no children Should render snapshot correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "#0FCFA6",
          "borderColor": "#000000",
          "borderRadius": 8,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "column",
        },
        Array [
          Object {
            "elevation": 2,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 4,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 10,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "margin": 8,
          },
          Array [
            Object {},
          ],
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "display": "flex",
            "flexWrap": "nowrap",
            "margin": 8,
            "width": "80%",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 24,
              "textAlign": "center",
            },
          ]
        }
      >
        Test title
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 14,
              "lineHeight": 18,
              "marginTop": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Test sub text
      </Text>
    </View>
  </View>
</View>
`;

exports[`components >> Select Box Given select box has a variant of vertical Given select box is NOT selected and has children Should render snapshot correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "#f4f5f5",
          "borderColor": "#000000",
          "borderRadius": 8,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "column",
        },
        Array [
          Object {
            "elevation": 2,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 4,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 10,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "margin": 8,
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "display": "flex",
            },
          ]
        }
      >
        TEST CHILDREN
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "display": "flex",
            "flexWrap": "nowrap",
            "margin": 8,
            "width": "80%",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 24,
              "textAlign": "center",
            },
          ]
        }
      >
        Test title
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 14,
              "lineHeight": 18,
              "marginTop": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Test sub text
      </Text>
    </View>
  </View>
</View>
`;

exports[`components >> Select Box Given select box has a variant of vertical Given select box is NOT selected and has no children Should render snapshot correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "#f4f5f5",
          "borderColor": "#000000",
          "borderRadius": 8,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "column",
        },
        Array [
          Object {
            "elevation": 2,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 4,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 10,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "margin": 8,
          },
          Array [
            Object {},
          ],
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "display": "flex",
            "flexWrap": "nowrap",
            "margin": 8,
            "width": "80%",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 24,
              "textAlign": "center",
            },
          ]
        }
      >
        Test title
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 14,
              "lineHeight": 18,
              "marginTop": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Test sub text
      </Text>
    </View>
  </View>
</View>
`;
